### PR TITLE
Remove all gamemodes except Dynamic

### PR DESCRIPTION
**Pull request by Mothblocks**
## About The Pull Request
Requested by @optimumtact.

Removes all `/datum/game_mode` except dynamic. Eventually, all of `mode` and `game_mode` will be removed, and Dynamic will become an ingrained system. Every single other gamemode was unmaintained at best and poisoned other code at worst. Currently all tg servers run 24/7 Dynamic, so the time to act is now.

To admins: If you want to force modes like extended, use the "Force Roundstart Ruleset" option in the game panel instead.

Pretty much completely untested 😉 
On Do Not Merge until proper testing is done.

TODO:
- [x] Test uplinks
- [x] Test candidate polling
- [x] Remove one-click antag, put ERT in its own verb, and N.E.R.D. in Secrets.
- [x] Test merge.

:cl:
del: One-click antagonist has been removed. Consider executing midround rulesets or using the traitor panel instead.
admin: Calling an ERT is now its own verb.
admin: The N.E.R.D. has been moved to secrets.
/:cl:

### CODE REVIEW

**Comment by wesoda25** [comment]
How many people are really going to know what you mean by "one-click antagonist"?

**Comment by Mothblocks** [comment]
That's the best proof of how useless it is, really.

**Comment by CRITAWAKETS** [comment]
oh man dont remove nothing, i love playing around with nothing

#removedontimprove

**Comment by SuperNovaa41** [comment]
Please don't remove monkey....

**Comment by Mothblocks** [comment]
monkey is already a dynamic ruleset u can force as an admin

**Comment by Mothblocks** [comment]
same with meteors for whatever reason actually

**Comment by nianjiilical** [comment]
i feel like one click antag still has use for when you want to inject a specific amount of a specific antag into a round, unless im missing some other button that has the same effect

**Comment by Mothblocks** [comment]
It's slightly more convenient yeah, but it's not worth the added maintainability cost. Instead, consider just executing midround rulesets to inject sleeper agents, and press the button a lot.

**Comment by Mothblocks** [comment]
Ready to go.

**Comment by LemonInTheDark** [comment]
`One-click antagonist has been removed` nooooooo we just fixed thatt

**Comment by Mothblocks** [comment]
revs still didnt work on it i think

**Comment by MrStonedOne** [comment]
Can you make the repo default rule set be extended. Or optionally leave extended. Or optionally close this pr and go home and think about your life.

Local testing with an actual game mode is cancer.

Is there an easy way to force extended at the code level? Our event servers will need this. Did we ever make the events define?

**Comment by Mothblocks** [comment]
> Local testing with an actual game mode is cancer.

If you have antags turned off, it'll just be extended.

> Is there an easy way to force extended at the code level? 

You can configure Dynamic to turn off all other rulesets and such, or give Extended a ridiculously high weight yes. Admins can explicitly force extended as well.

> Or optionally leave extended

Leaving it would just be kicking the can down the road--all of `game_mode` is planned to be killed.

**Comment by Fikou** [comment]
PLEASE, WE NEED SPEEDY REVOLUTION

**Comment by Iamgoofball** [comment]
:+1: hype af

**Comment by Iamgoofball** [comment]
FUCK WRONG BUTTON

**Comment by Mothblocks** [comment]
😛

**Comment by Iamgoofball** [comment]
wish they'd add a fucking confirm box to that holy shit github

File: code/game/gamemodes/game_mode.dm [hunk 1]
```diff
@@ -591,25 +591,11 @@
...
-		return 0
-	if(!CONFIG_GET(flag/use_age_restriction_for_jobs))
-		return 0
-	if(!isnum(C.player_age))
-		return 0 //This is only a number if the db connection is established, otherwise it is text: "Requires database", meaning these restrictions cannot be enforced
-	if(!isnum(enemy_minimum_age))
-		return 0
-
-	return max(0, enemy_minimum_age - C.player_age)
+	// MOTHBLOCKS TODO: kill
```
Comment (Mothblocks) [inline code/game/gamemodes/game_mode.dm:594]: hold on

**Comment by rdragan** [comment]
> wish they'd add a fucking confirm box to that holy shit github

wyci

**Comment by MrStonedOne** [comment]
> Can you make the repo default rule set be extended.

**Comment by Mothblocks** [comment]
It's *possible*, sure, but I don't see what's so objectionable about just not having antags on--this'll be fundamentally equivalent to extended.

**Comment by KathrinBailey** [comment]
> Leaving it would just be kicking the can down the road--all of `game_mode` is planned to be killed.

Out of curiosity are these and other plans out to see anywhere, or is all discussion of said plans kept only in the TG maintainer James Bond-esque supervillain lair? If it's the supervillain lair, do all maintainers have a trap door under their seat at the meeting table and does Oranges wear an eyepatch?

**Comment by hugmeorelse** [comment]
Hi, will it be possible to configure what dynamic mode will run? The dynamic.json config file seems woefully unprepared for that task.

**Comment by KathrinBailey** [comment]
Also does dynamic not give antag to those antag-banned and spawn antags that are disabled on TG?
Whenever we use dynamic on our downstream it'll just give traitor to people who don't even have it enabled, and sometimes to people banned from traitor.

**Comment by ShiftyRail** [comment]
>To admins: If you want to force modes like extended, use the "Force Roundstart Ruleset" option in the game panel instead.

No, you don't need to that. You would need to use the dynamic mode option "Forced extended". This ruleset doesn't exactly work, unfortunately.
From a coder at the birthplace of dynamic mode.

**Comment by ExcessiveUseOfCobblestone** [comment]
> Will it be possible to configure what dynamic mode will run? The dynamic.json config file seems woefully unprepared for that task, unless I'm missing something which is entirely possible!

What do you mean dynamic mode? You can configure each rule set pretty extensively

**Comment by Mothblocks** [comment]
> Out of curiosity are these and other plans out to see anywhere, or is all discussion of said plans kept only in the TG maintainer James Bond-esque supervillain lair? If it's the supervillain lair, do all maintainers have a trap door under their seat at the meeting table and does Oranges wear an eyepatch?

There's nothing formal, just chatter in the coding general of tgcord.

> Whenever we use dynamic on our downstream it'll just give traitor to people who don't even have it enabled, and sometimes to people banned from traitor.

It shouldn't, no. I have basically all antags off on tg for more than a year and have never rolled them. Banning from traitor should be similar, though I can't speak for that since I'm not banned from anything, but just that I feel like I would've heard it on the grape vine as an admin by now.

**Comment by Mothblocks** [comment]
@hugmeorelse I'd also like to know what you mean by "dynamic mode"--the whole point is it runs multiple things!

If you want to force one mode always, you can set the weight of everything else to zero. If this seems like a pain to you...that's because dynamic is designed to explicitly do the *opposite* of this task.

**Comment by EgorDinamit** [comment]
First, they came for intents;
Then, they came for the game modes...

**Comment by hugmeorelse** [comment]
> > Will it be possible to configure what dynamic mode will run? The dynamic.json config file seems woefully unprepared for that task, unless I'm missing something which is entirely possible!
> 
> What do you mean dynamic mode? You can configure each rule set pretty extensively

To clarify, I am asking if there will be config files for it, so it can be adjusted on the fly.

Also, I have no idea what you mean by "what do you mean dynamic mode?" I'm just talking about this game mode or ruleset, whatever you want to call it.

**Comment by Mothblocks** [comment]
Dynamic's configuration is vaster than anything else we provide, and is extensively documented: https://github.com/tgstation/tgstation/blob/master/code/game/gamemodes/dynamic/readme.md

**Comment by optimumtact** [comment]
all mso is asking is that you make the default config not spawn any antags

**Comment by Mothblocks** [comment]
Yeah, going to do that when I can.

**Comment by PKPenguin321** [comment]
Is there still a way to force "secret extended" (as in, prevent all antags/gamemode-specific things from spawning) in order to run events with this?

**Comment by Mothblocks** [comment]
Yes.

As an admin, "Force Roundstart Ruleset" -> Extended. This will still start the blue alert and whatnot.

**Comment by Stalkeros2** [comment]
Literally removing the game at this point.

**Comment by Mothblocks** [comment]
This has been test merged on Sybil, and nothing exploded.

File: code/datums/mind.dm [hunk 2]
```diff
@@ -341,14 +331,6 @@
 		if(O)
 			O.unlock_code = null
 
-/datum/mind/proc/remove_all_antag() //For the Lazy amongst us.
```
Comment (Mothblocks) [inline code/datums/mind.dm:344]: Yo?

Comment (LemonInTheDark) [reply code/datums/mind.dm:344]: don't think it

**Comment by TheVekter** [comment]
Any reason behind removing one-button antag? How would I go about making a random antag without it?

**Comment by Mothblocks** [comment]
It was extreeeeemely old and game-mode code dependent, and when I asked in bus people said they didn't use it.

> How would I go about making a random antag without it?

Depends on the antag. For traitor, just execute midround sleeper agents. For revs, execute latejoin provocateur (and the next person will become a headrev). Cult is the trickiest one, but across the board people said that they don't really like creaitng midround cults anyway, so at that point you're better off executing a roundstart ruleset.

**Comment by MrStonedOne** [comment]
I think this pr should be delayed until you make a proper replacement.

1: waiting for the next latejoin to do the thing is not the same as mid round antag, which injected it in a random person currently in the round.

2: you don't have a solution for all the antag types. 

3: you keep pigeonholing everything into the "normal" round, but admins are and should be free to create custom wacky wild *abnormal* rounds.

4: the removal of one click antag is unrelated to removing game modes, and should be its own pr.

**Comment by optimumtact** [comment]
> 4: the removal of one click antag is unrelated to removing game modes, and should be its own pr.
it's not because the code is hosted by the same set of datums

**Comment by JJRcop** [comment]
When Dynamic was first introduced I was hoping it would eventually end up like this, very exciting! Ever since datum antags came into the picture I wanted game_mode to die.

**Comment by JJRcop** [comment]
When Dynamic was first introduced I was hoping it would eventually end up like this, very exciting! Ever since datum antags came into the picture I hoped game_mode to die.

**Comment by Mothblocks** [comment]
Waiting on #58625 before fixing uplink stuff.

**Comment by Mothblocks** [comment]
If I remember right, MSO said over Discord that he is okay removing one-click antag, but that we should eventually replace it with a button that opens up the Traitor Panel of a random player with the preference selected, which I think is fine.

**Comment by LemonInTheDark** [comment]
YOLO

**Pull request merged**